integration_tests/src/main/python/delta_lake_low_shuffle_merge_test.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pyspark.sql.functions as f
+import pytest
+import string
+
+from asserts import *
+from data_gen import *
+from delta_lake_utils import *
+from marks import *
+from pyspark.sql.types import *
+from spark_session import *
+
+# Databricks changes the number of files being written, so we cannot compare logs
+num_slices_to_test = [10] if is_databricks_runtime() else [1, 10]
+
+delta_merge_enabled_conf = copy_and_update(delta_writes_enabled_conf,
+                                           {
+                            "spark.rapids.sql.command.MergeIntoCommand": "true",
+                            "spark.rapids.sql.command.MergeIntoCommandEdge": "true",
+                            "spark.rapids.sql.delta.lowShuffleMerge.enabled": "true"
+                                            })
+
+def make_df(spark, gen, num_slices):
+    return three_col_df(spark, gen, SetValuesGen(StringType(), string.ascii_lowercase),
+                        SetValuesGen(StringType(), string.ascii_uppercase), num_slices=num_slices)
+
+def delta_sql_merge_test(spark_tmp_path, spark_tmp_table_factory, use_cdf,
+                         src_table_func, dest_table_func, merge_sql, check_func,
+                         partition_columns=None):
+    data_path = spark_tmp_path + "/DELTA_DATA"
+    src_table = spark_tmp_table_factory.get()
+    def setup_tables(spark):
+        setup_delta_dest_tables(spark, data_path, dest_table_func, use_cdf, partition_columns)
+        src_table_func(spark).createOrReplaceTempView(src_table)
+    def do_merge(spark, path):
+        dest_table = spark_tmp_table_factory.get()
+        read_delta_path(spark, path).createOrReplaceTempView(dest_table)
+        return spark.sql(merge_sql.format(src_table=src_table, dest_table=dest_table)).collect()
+    with_cpu_session(setup_tables)
+    check_func(data_path, do_merge)
```

Comment:
This and other stuff seems identical to the versions in delta_lake_merge_test?  Not sure this should be a full-blown separate test suite from delta_lake_merge_test, especially with low-shuffle being on by default.  That means delta_lake_merge_test is actually a low shuffle merge test, since it's on by default, and we're not testing when it's off.  Better to parameterize the existing merge tests in delta_lake_merge_test with low- and non-low- confs than build a separate suite here.  Confs for parameterization can be based on the platform, so we avoid trying to test low-shuffle merge on platforms that do not support it.

Reply:
Is this still going to get fixed?  There's needless duplication of lots of test code here that I don't see why we need to postpone.  It's trivial to fix.